### PR TITLE
Database and API testing

### PR DESCRIPTION
**Pull request by jeffbw13**
Includes all the api except the reservation post

### CODE REVIEW

File: database/Mapper.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+let Mapper = mongoose.model('Mapper', mapperSchema);
+
+var getAll = () => {
+  return new Promise((resolve, reject) => {
+    Mapper.find()
+      .exec((err, maps) => {
+        resolve(maps);
+      });
+  });
```
Comment (happylolo) [inline database/Mapper.js]: maybe better to add reject() to handle err

File: database/Mapper.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+var getOne = (restaurantId) => {
+  return new Promise((resolve, reject) => {
+    Mapper.find({ restaurant_id: restaurantId })
+      .exec((err, map) => {
+        resolve(map);
+      });
+  });
```
Comment (happylolo) [inline database/Mapper.js]: same here.

File: database/Reservation.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+// function to make a reservation.  or not.
+var make = (restId, name, time) => {
+  //  1. get all reservations for this restaurant, date.  Not very efficient.
+  //  2. pare down to overlapping dateTimes
+  //  3. check if <name> has an overlapping dateTime.  If so, decline
+  //  4. if count of existing reservation dateTime overlaps (array created in step 2) is already at
+  //       max, decline the new reservation
+  //  5. else, add the reservation to the database!
+  getByDate(name, time)
```
Comment (happylolo) [inline database/Reservation.js]: From the comment above (Line 25), it looks like the getByDate won't work? so I'm assuming this `make` function also won't work right now? Maybe you could add some unit tests later to verify it after the code is complete.

File: database/Reservation.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    }
+    if (nameMatch || overlaps.length >= AVAIL_TABLES) {
+      //  return declined reason
+    } else {
+      //  add to the database
+    }
+  })
+  .catch((err) => {
+    //console.log('Error:', err);
+    //res.end();
```
Comment (happylolo) [inline database/Reservation.js]: why is this commented out?

File: server/index.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+app.get('/reservation/all', function (req, res) {
+  Reservation.getAll()
+  .then((reservations) => {
+    res.write(JSON.stringify(reservations));
+    res.end();
+  })
+  .catch((err) => {
+    console.log('Error: ', err);
+    res.end();
+  });
```
Comment (happylolo) [inline server/index.js]: not sure what status code is returned to the client here?

File: server/index.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+  Reservation.make(restId, name, time)
+  .then((confirmation) => {
+    res.write(JSON.stringify(confirmation));
+    res.end();
+  })
+  .catch((err) => {
+    console.log('Error:', err);
+    res.end();
+  });
```
Comment (happylolo) [inline server/index.js]: same here

File: server/index.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+app.get('/mapper/all', function (req, res) {
+  Mapper.getAll()
+    .then((maps) => {
+      res.write(JSON.stringify(maps));
+      res.end();
+    })
+    .catch((err) => {
+      console.log('Error:', err);
+      res.end();
+    });
```
Comment (happylolo) [inline server/index.js]: same here.

File: server/index.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+      res.write(JSON.stringify(map));
+      res.end();
+    })
+    .catch((err) => {
+      console.log('Error: ', err);
+      res.end();
+    });
+});
+
+let port = 3002;
```
Comment (happylolo) [inline server/index.js]: aha! I use 3001 so our ports won't conflict with each other.

**Comment by jeffbw13** [comment]
Merged in another branch

**Pull request closed without merging**